servicebroker/lib/index.js

```diff
@@ -14,6 +15,28 @@ var app = express();
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
 
+var auth = function (req, res, next) {
+  
+  function unauthorized(res) {
+    res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+    return res.sendStatus(401);
+  };
+
+  var user = basicAuth(req);
+  if (!user || !user.name || !user.pass) {
+    return unauthorized(res);
+  };
+
+  if (user.name === settings.user && user.pass === settings.password) {
+    return next();
+  } else {
+    return unauthorized(res);
+  };
+  next();
+};
+
+app.use(auth);
```

Comment:
Yes, using middleware of express, we do the auth first before process the body.  I added console log in code to show the sequence, and result is:
> node lib/index.js

processing the auth first
processing the body later